images/cassandra/rootfs/initialize.sh

```diff
@@ -9,4 +9,6 @@ fi
 
 sed -i 's/REPLICATION_FACTOR/'"${REPLICATION_FACTOR}"'/g' /init.cql
 
-cqlsh -f /init.cql cassandra.default.svc.cluster.local
+if ! cqlsh cassandra.default.svc.cluster.local -k storage -e "select * from storage.block"; then
```

Comment:
is this to ensure that `init.cql` is not run on a database where the select returns something?
can it return an empty dataset?
can the cql script be written so it creates or updates?
The reason is that basing this decision on `cqlsh` failing or succeeding is a bit dangerous - it does not have enough context to make a decision about whether to initialize (or does it?)

Reply:
Agree with @a-palchikov. I debugged this issue last Friday and wrote an extent summary of it at AOP-1609.
The bottom line is:
```
There are 3 things to do here:
1. Make sure cassandra is responding correctly before executing the command
2. Increate cql command timeout
3. Make the cql script or the shell script to not fail if the assets exist
```

Reply:
The real solution is to make sure that `init.cql` is idempotent.

Reply:
`CREATE TABLE IF NOT EXISTS keyspace_name.table_name` can be the solution for every statement.